pinot-core/src/main/java/org/apache/pinot/core/plan/GlobalPlanImplV0.java

```diff
@@ -45,11 +46,15 @@ public PlanNode getPlanNode() {
   }
 
   @Override
-  public DataTable execute() {
+  public DataTable execute()
+      throws TimeoutException {
     long startTime = System.currentTimeMillis();
     InstanceResponseOperator instanceResponseOperator = _instanceResponsePlanNode.run();
     long endTime1 = System.currentTimeMillis();
     LOGGER.debug("InstanceResponsePlanNode.run() took: {}ms", endTime1 - startTime);
+    if (endTime1 > _instanceResponsePlanNode._queryContext.getEndTimeMs()) {
+      throw new TimeoutException("Timeout after plan generation");
```

Comment:
(nit) suggest changing this to `"Query timed out while generating physical execution plan"`

Reply:
Done